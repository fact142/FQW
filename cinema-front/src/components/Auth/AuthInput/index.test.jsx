import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { AuthInput } from './index';

const testingValue = 'test value';
const testingPlaceholder = 'test placeholder'
const onChange = jest.fn();

const TestingFormInput = () => (
  <AuthInput value={testingValue} onChange={onChange} placeholder={testingPlaceholder}/>
);

describe('FormInput component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<TestingFormInput />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should call onChange when the value change ', () => {
    const { getByTestId } = render(<TestingFormInput />);
    const input = getByTestId('form-input');
    userEvent.type(input, '1');
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
