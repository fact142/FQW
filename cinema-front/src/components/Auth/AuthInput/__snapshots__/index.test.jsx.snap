// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="form"
  >
    <form />
  </div>
</DocumentFragment>
`;
